azurerm/internal/services/network/resource_arm_web_application_firewall_policy.go

```diff
@@ -48,7 +48,7 @@ func resourceArmWebApplicationFirewallPolicy() *schema.Resource {
 
 			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
 
-			"custom_rules": {
+			"custom_rule": {
```

Comment:
There can be no existing users of the resource as it does not work, as described in #5354. I thought that before making the resource actually work, this would be a good chance to align naming with other resources. The existing docs even mention a `custom_rule` block:

![image](https://user-images.githubusercontent.com/425633/77225636-fd653900-6b70-11ea-8c3b-15dd7193d56c.png)

The same goes for `match_condition` and `match_variable`.

A single block with a plural name would indicate that the block configures all such elements, e.g. custom rules, whereas multiple blocks with a singular name indicate that they each configure such an element.

Reply:
I see now that the `ManagedRules` field was introduced in #5004 (2e69260f3a) between dependency `github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network` and `github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network`, but the `azurerm_web_application_firewall_policy` resource was not updated to reflect this new required field.

This of course means that there will be existing users of this resource that have used the `custom_rules` name for the property before the `ManagedRules` field was introduced, so I suppose I'll revert this change after all, and update the docs to all use the actual property names instead of the singular forms.

Reply:
Fixed in 7aeb41b5f.